src/components/Contacts/ContactDetails/ContactDetailsHeader/DeleteContactModal/DeleteContactModal.tsx

```diff
@@ -12,37 +17,178 @@ import {
+    // If any data is not editable, this means it was created by a third party.
+    // Which will only recreate the data after deleting it on MPDX.
+    // To prevent this confusion, we do not allow a contact to be deleted if it has non editable data.
+
+    const addressSources = new Set<string>();
+    const emailSources = new Set<string>();
+    const phoneSources = new Set<string>();
+
+    contactSources.addresses?.nodes.forEach((address) => {
+      if (!isEditableSource(address.source)) {
+        addressSources.add(address.source);
+      }
+    });
+
+    contactSources.people?.nodes?.forEach((person) => {
+      person.emailAddresses.nodes.forEach((email) => {
+        if (!isEditableSource(email.source)) {
+          emailSources.add(email.source);
+        }
+      });
+      person.phoneNumbers.nodes.forEach((phone) => {
+        if (!isEditableSource(phone.source)) {
+          phoneSources.add(phone.source);
+        }
+      });
+    });
+
+    return {
+      canDeleteWithoutIssues:
+        isEditableSource(contactSources.source ?? '') &&
+        !addressSources.size &&
+        !emailSources.size &&
+        !phoneSources.size,
+      contactSource: sourceToStr(t, contactSources.source),
+      addressSources: [...addressSources].map((source) =>
+        sourceToStr(t, source),
+      ),
+      emailSources: [...emailSources].map((source) => sourceToStr(t, source)),
+      phoneSources: [...phoneSources].map((source) => sourceToStr(t, source)),
+    };
+  }, [contactSources]);
+
   return (
     <Modal
       isOpen={open}
       title={t('Delete Contact')}
       handleClose={() => setOpen(false)}
     >
       <DialogContent dividers>
-        <DialogContentText>
-          {t(
-            'Are you sure you want to permanently delete this contact? Doing so will permanently delete this contacts information, as well as task history. This cannot be undone. If you wish to keep this information, you can try hiding this contact instead.',
-          )}
-        </DialogContentText>
+        {dataInfo.canDeleteWithoutIssues ? (
+          <Typography>
+            {t(
+              `Are you sure you want to permanently delete this contact? Doing so will permanently delete this contacts information, as well as task history. This cannot be undone. If you wish to keep this information, you can try hiding this contact instead.`,
+            )}
+          </Typography>
+        ) : (
+          <>
+            <Typography>
+              {t(
+                `Be cautious when deleting this contact, as its data may sync with Donation Services or other third-party systems. Deleting the contact will not remove it from those systems; consider hiding the contact instead.`,
+              )}
+            </Typography>
+            <Typography fontWeight="bold">
+              {t(
+                `For contacts originating from Siebel or Donor Hub, email Donation Services to request deletion.`,
```

Comment:
Most users won't know what Siebel is, and "US Donation Services" is what will show up in the list. Also, we can remove the space from "Donor Hub" to match what `sourceToStr` returns.

Is it possible to include the Donation Services email address in this message? Maybe that's bad though if it shows up for non-Cru orgs.

```suggestion
                `For contacts originating from Donation Services or DonorHub, email Donation Services to request deletion.`,
```

Reply:
Yeah, that works. I wouldn't want to delay this PR any further.